crates/uv/tests/pip_compile.rs

```diff
@@ -3096,6 +3096,56 @@ fn compile_editable() -> Result<()> {
     Ok(())
 }
 
+/// If an editable is repeated, it should only be built once.
+#[test]
+fn deduplicate_editable() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str(indoc! {r"
+        -e file://../../scripts/packages/black_editable
+        -e file://../../scripts/packages/black_editable
+        -e file://../../scripts/packages/black_editable[dev]
```

Comment:
Can you add test coverage for an absolute path that resolves to the same as a relative path?

Should we add handling for symbolic links?

Reply:
I'd prefer not to merge symlinks here, but I can add an absolute path test, yes.

Reply:
I mean, we _could_ merge symlinks, it's not difficult, I'm just not sure if it's correct. I guess in theory they _must_ resolve to the same package, so maybe it's totally fine...